### ISSUE TITLE
bug: AxiosRequestConfig.proxy should allow type false

### ISSUE THREAD

**Issue by dhcmrlchtdj** [axios/axios#1154]
#### Summary

I can't create an instance with `axios.create({proxy: false})`.
tsc report 

```
error TS2345: Argument of type '{ proxy: boolean; }' is not assignable to parameter of type 'AxiosRequestConfig | undefined'.
  Type '{ proxy: boolean; }' is not assignable to type 'AxiosRequestConfig'.
    Types of property 'proxy' are incompatible.
      Type 'boolean' is not assignable to type 'AxiosProxyConfig | undefined'.
```

The typing should be `AxiosProxyConfig | false`, however `false` is missing at current version.

https://github.com/axios/axios/blob/fbb5d4d0d1d4dd1cdbb02f30d05738e0b8e5316b/index.d.ts#L43
https://github.com/axios/axios/blob/fbb5d4d0d1d4dd1cdbb02f30d05738e0b8e5316b/lib/adapters/http.js#L85

#### Context

- axios version: *v0.17.0*

**Comment by nickuraltsev** [comment]
Right! I'd be happy to accept a PR.

**Comment by nickuraltsev** [comment]
Fixed via #1177
